tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -1605,24 +1604,182 @@ def _test_memory_based_scheduling_enabled_false(
     assert_that(slurm_commands.get_node_attribute("queue1-st-ondemand1-i1-1", "Memory")).is_equal_to("3891")
     assert_that(slurm_commands.get_node_attribute("queue1-dy-ondemand1-i3-1", "Memory")).is_equal_to("31129")
 
-    # TODO: Add functional tests for memory-based scheduling
+    # Upload files for memory allocation tests
+    remote_command_executor._copy_additional_files(
+        [
+            str(test_datadir / "memory_allocation_chars.c"),
+        ],
+    )
+
+    # Compile C program to test memory allocations
+    remote_command_executor.run_remote_command("gcc memory_allocation_chars.c")
+    assert_that(remote_command_executor.run_remote_command("ls ./a.out").stdout).contains("a.out")
```

Comment:
If `a.out` is not there, the `ls` command would fail, so the assertion on `contains` is redundant.
You could simply run the `ls` command to verify that it is there